### PR TITLE
Update base job template validation to only run on work pool update and creation

### PR DESCRIPTION
**Pull request by desertaxle**
Closes https://github.com/PrefectHQ/prefect/issues/10458

We fixed a bug allowing misconfigured base job templates to be created for work pools. After fixing that bug, some users cannot read their work pools due to a misconfigured base job template. This PR moves validation to only work pool update and creation to unblock work pool reads. Users must fix their misconfigured base job templates to update a work pool with a misconfigured base job template.

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

File: tests/server/api/test_workers.py [hunk 1]
```diff
@@ -465,6 +503,15 @@ async def test_read_work_pools_with_offset(self, client, session):
...
 
+    async def test_read_work_pool_with_work_pool_that_fails_validation(
+        self,
+        client,
+        invalid_work_pool,
+    ):
+        response = await client.post("/work_pools/filter")
+        assert response.status_code == 200
+        assert len(response.json()) == 4
+
```
Comment (serinamarie) [inline tests/server/api/test_workers.py:514]: Would it be good to test that updating `invalid_work_pool` both properly and improperly causes `WorkPoolUpdate` to work as expected?

Comment (desertaxle) [reply tests/server/api/test_workers.py:514]: Definitely! But there are some existing base job template validation tests for work pool creation and update:
- [test_create_work_pool_template_validation_missing_variables](https://github.com/PrefectHQ/prefect/blob/main/tests/server/api/test_workers.py#L124)
- [test_create_work_pool_template_validation_missing_nested_variables](https://github.com/PrefectHQ/prefect/blob/main/tests/server/api/test_workers.py#L124)
- [test_update_work_pool_template_validation_missing_variables](https://github.com/PrefectHQ/prefect/blob/main/tests/server/api/test_workers.py#L345)
- [test_update_work_pool_template_validation_missing_nested_variables](https://github.com/PrefectHQ/prefect/blob/main/tests/server/api/test_workers.py#L386)
These tests make sure we're maintaining the existing functionality for those routes.

Comment (serinamarie) [reply tests/server/api/test_workers.py:514]: Great! Those seem like they're covering it then

File: tests/cli/test_deploy.py [hunk 2]
```diff
@@ -623,7 +623,7 @@ async def test_project_deploy_with_no_prefect_yaml(self, project_dir, work_pool)
             expected_output_contains=[
                 "Your Prefect workers will attempt to load your flow from:",
                 "To see more options for managing your flow's code, run:",
-                "$ prefect project recipes ls",
+                "$ prefect init",
```
Comment (serinamarie) [inline tests/cli/test_deploy.py:626]: If merged this will supercede https://github.com/PrefectHQ/prefect/pull/10480

**Review by serinamarie — APPROVED** [review]
This looks good to me!

**Pull request merged**